### PR TITLE
Fixed: Add links to DAGRun / DAG / Task in templates-ref.rst

### PR DESCRIPTION
**Pull request by fritz-astronomer**
Reverts apache/airflow#33010 , continuation of apache/airflow#32245

~Need to figure out how to correctly refer to Airflow source code objects with `.rst`~
The docs readme has a lovely section on the `:class:` reference, which I used here
https://github.com/apache/airflow/tree/main/docs#cross-referencing-syntax

### CODE REVIEW

File: docs/apache-airflow/templates-ref.rst [hunk 1]
```diff
@@ -47,16 +47,16 @@ Variable                                    Type                  Description
...
                                             | ``None``            | Added in version 2.2.
-``{{ prev_data_interval_end_success }}``    `pendulum.DateTime`_  | End of the data interval of the prior successful DAG run.
+``{{ prev_data_interval_end_success }}``    `pendulum.DateTime`_  | End of the data interval of the prior successful `DAG Run`_.
                                             | ``None``            | Added in version 2.2.
-``{{ prev_start_date_success }}``           `pendulum.DateTime`_  Start date from prior successful DAG run (if available).
+``{{ prev_start_date_success }}``           `pendulum.DateTime`_  Start date from prior successful `DAG Run`_ (if available).
                                             | ``None``
-``{{ dag }}``                               DAG                   The currently running DAG.
-``{{ task }}``                              BaseOperator          | The currently running task.
+``{{ dag }}``                               DAG                   The currently running `DAG`_. You can read more about DAGs in :doc:`DAGs <core-concepts/dags>`.
```
Comment (uranusjr) [inline docs/apache-airflow/templates-ref.rst:56]:
```suggestion
``{{ dag }}``                               DAG                   The currently running :doc:`DAG <core-concepts/dags>`.
```

Would this work?

Comment (uranusjr) [inline docs/apache-airflow/templates-ref.rst]:
```suggestion
``{{ dag }}``                               DAG                   The currently running :class:`~airflow.models.dag.DAG`. You can read more about DAGs in :doc:`DAGs <core-concepts/dags>`.
```

I think you can just do this without the indirection below. The indirect syntax is more useful if you want to link to `DAG` several times. It’s just preference though.

Comment (fritz-astronomer) [reply docs/apache-airflow/templates-ref.rst]: The indirect wasn't intentional (though, it does make sense, as you mention if you reference it multiple times) - I was just copying an example elsewhere in the docs.

**Comment by fritz-astronomer** [comment]
My docker segfaulted trying to run breeze, so I can't test this again :/

**Comment by fritz-astronomer** [comment]
This is good to go 👍 I was able to test with breeze. `DagRun` doesn't fill with a link, for some reason that isn't rendered in the Airflow Docs, but it's just a non-clickable classpath, which is fine in my book. The other links (baseoperator, task, dag) render correctly and I tested them.

I'd like to rename `airflow.models.dag.DAG` to just `DAG`, but ```:class:`foo <foo>` ``` doesn't seem to work, so I'm happy shipping this as-is

**Comment by uranusjr** [comment]
> I'd like to rename `airflow.models.dag.DAG` to just `DAG`, but `` :class:`foo <foo>` `` doesn't seem to work, so I'm happy shipping this as-is

I think ``:class:`~airflow.models.dag.DAG` `` would work (the `~` at the front trims out the module path)

**Comment by fritz-astronomer** [comment]
Made the `~` change, and tested. Ready to merge 👍

**Review by uranusjr — APPROVED** [review]

**Pull request merged**